h/api/models/annotation.py

```diff
@@ -75,5 +80,53 @@ class Annotation(Base, mixins.Timestamps):
     #: Any additional serialisable data provided by the client.
     extra = sa.Column(pg.JSONB, nullable=True)
 
+    document_uris = sa.orm.relationship('DocumentURI',
+                                        viewonly=True,
+                                        foreign_keys='DocumentURI.uri_normalized',
+                                        primaryjoin='DocumentURI.uri_normalized == Annotation.target_uri_normalized')
+    documents = association_proxy('document_uris', 'document')
+
+    @hybrid_property
+    def target_uri(self):
+        return self._target_uri
+
+    @target_uri.setter
+    def target_uri(self, value):
+        self._target_uri = value
+        self._target_uri_normalized = text_type(uri.normalize(value), 'utf-8')
+
+    @hybrid_property
+    def target_uri_normalized(self):
+        return self._target_uri_normalized
+
+    @hybrid_property
+    def document(self):
+        uris = self.document_uris
```

Comment:
Given the association proxy we have above, this might be clearer as:

```python
def document(self):
    if self.documents:
        return self.documents[0]
```